cortex_xdr/plugin.spec.yaml

```diff
@@ -0,0 +1,232 @@
+    type: string
+    required: true
+    example: https://api-example.xdr.us.paloaltonetworks.com/
+  api_key_id:
+    title: API Key ID
+    description: The API Key ID shown in the API Keys table in settings. e.g. 1, 2, 3
+    type: int
+    required: true
+    example: 1
+  api_key:
+    title: API Key
+    description: The API Key that is generated when creating a new key
+    type: credential_secret_key
+    required: true
+    example: '1234123412341234asdfasdfasdfasdfasdf1234123412341234123412341234asdfasdfasdfasdfasdf123412341234123412341234asdfasdfasdfasdfasdf'
+  security_level:
+    title: Security Level
+    description: The Security Level of the key provided. This can be found in the API Key settings table in the Cortex XDR settings
+    type: string
+    required: true
+    example: Standard
+    default: Standard
+    enum:
+      - Advanced
+      - Standard
+actions:
+  get_endpoint_details:
+    title: Get Endpoint Details
+    description: Get information about an endpoint
+    input:
+      endpoint:
+        title: Endpoint
+        description: The endpoint to get information about. Endpoint will accept IPv4, hostnames, and endpoint IDs
+        type: string
+        required: true
+        example: 0123456abcdef12345abcde12345abcd
+    output:
+      total_count:
+        title: Total Count
+        description: Number of results found (max 100)
+        type: integer
+        required: true
+      endpoints:
+        title: Endpoints
+        description: Any endpoints that match the given endpoint information
+        type: "[]endpoint"
+        required: true
+  isolate_endpoint:
+    title: Isolate Endpoint
+    description: Isolate an endpoint
+    input:
+      endpoint:
+        title: Endpoint
+        description: Endpoint to take isolation action on. This can be IPv4, hostnames, and endpoint IDs
+        type: string
+        required: true
+        example: 0123456abcdef12345abcde12345abcd
+      isolation_state:
+        title: Isolation State
+        description: True to isolate host, false to unisolate a host
+        type: boolean
+        required: true
+        default: True
+        example: True
+      whitelist:
+        title: Whitelist
+        description: This list contains a set of devices that should not be blocked. This can include IPs, hostnames, and device IDs
+        type: "[]string"
+        required: false
+        default: []
+        example: ["198.51.100.100", "hostname123", "225494730938493804"]
+    output:
+      result:
+        title: Result
+        description: The result of the isolation request
+        type: isolation_result
+        required: true
+
+triggers:
+  get_incidents:
+    title: Get Incidents
+    description: Get Incidents
+    input:
+      time_field:
+        title: Time Field
+        description: Which time field to filter and sort on
+        type: string
+        required: true
+        enum:
+          - "modification_time"
+          - "creation_time"
```

Comment:
It would look more friendly to the user to see "Modification Time" and "Creation Time" - then transform these values to what the API requires within the code.

Reply:
Removed...this doesn't really serve a purpose in the trigger. I'm not sure why you'd sort trigger events.